src/Target.cpp

```diff
@@ -164,19 +234,23 @@ Target calculate_host_target() {
         }
         if ((info2[1] & avx512) == avx512) {
             initial_features.push_back(Target::AVX512);
+            // FIXME: port to family/model -based detection.
             if ((info2[1] & avx512_knl) == avx512_knl) {
```

Comment:
These avx512_foo features don't mean "tune for processor foo, which happens to have avx512", they mean "enable the avx512 instructions that exist on this processor". It was our crude attempt at making sense of the many overlapping avx512 feature sets.

So I think it's a bit of a misnomer to call the new feature AVX2_znver3. I'd just make it Znver3

Reply:
Aha, that is an interesting context, thanks!

But still, `avx512_foo` does result in `-mcpu=foo`, so i would say my comment isn't entirely wrong.

> So I think it's a bit of a misnomer to call the new feature AVX2_znver3. I'd just make it Znver3

Yay :)